File: Output/javascript/medium/EvaluateReversePolishNotation.js
/**https://leetcode.com/problems/evaluate-reverse-polish-notation/ */
//You are given an array of strings tokens that represents an arithmetic expression in a Reverse Polish Notation.
//Evaluate the expression. Return an integer that represents the value of the expression.


/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    let stack = [];
    for(let i = 0; i < tokens.length; i++){
        if(tokens[i] == '+'){
            let a = stack.pop();
            let b = stack.pop();
            stack.push(a + b);
        }
        else if(tokens[i] == '-'){
            let a = stack.pop();
            let b = stack.pop();
            stack.push(b - a);
        }
        else if(tokens[i] == '*'){
            let a = stack.pop();
            let b = stack.pop();
            stack.push(a * b);
        }
        else if(tokens[i] == '/'){
            let a = stack.pop();
            let b = stack.pop();
            stack.push(Math.trunc(b / a));
        }
        else{
            stack.push(parseInt(tokens[i]));
        }
    }
    return stack.pop();
    
};

    

File: Tool/PreUtil/src/output/java/medium/EvaluateReversePolishNotation.java
/**https://leetcode.com/problems/evaluate-reverse-polish-notation/ */
//You are given an array of strings tokens that represents an arithmetic expression in a Reverse Polish Notation.
//Evaluate the expression. Return an integer that represents the value of the expression.


class EvaluateReversePolishNotation {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for(String s : tokens){
            if(s.equals("+")){
                int a = stack.pop();
                int b = stack.pop();
                stack.push(a + b);
            }else if(s.equals("-")){
                int a = stack.pop();
                int b = stack.pop();
                stack.push(b - a);
            }else if(s.equals("*")){
                int a = stack.pop();
                int b = stack.pop();
                stack.push(a * b);
            }else if(s.equals("/")){
                int a = stack.pop();
                int b = stack.pop();
                stack.push(b / a);
            }else{
                stack.push(Integer.parseInt(s));
            }
        }
        return stack.pop();
        
        
        
    }
}

    